ambry-store/src/main/java/com.github.ambry.store/Log.java

```diff
@@ -308,6 +330,19 @@ private LogSegment checkArgsAndGetFirstSegment(long segmentCapacity) throws IOEx
   }
 
   /**
+   * @return {@code true} if the data directory contains segments that aren't old-style single segment log files.
+   * This should be run after a log is initialized to get the actual log segmentation state.
+   * @throws IOException
+   */
+  private boolean isExistingLogSegmented() throws IOException {
+    File[] segmentFiles = new File(dataDir).listFiles(LogSegmentNameHelper.LOG_FILE_FILTER);
```

Comment:
instead of trying to read from disk, what do you think of this implementation (since this is called after init):
`return !segmentsByName.firstKey().isEmpty()`?

Reply:
Yep this is a better way to do this. Thanks for the suggestion